src/Symfony/Contracts/Service/Attribute/SubscribedService.php

```diff
@@ -22,12 +27,21 @@
 #[\Attribute(\Attribute::TARGET_METHOD)]
 final class SubscribedService
 {
+    /** @var object[] */
+    public array $attributes;
+
     /**
-     * @param string|null $key The key to use for the service
-     *                         If null, use "ClassName::methodName"
+     * @param string|null       $key           The key to use for the service
+     * @param class-string|null $type          The service class
+     * @param bool              $nullable      Whether the service is optional
+     * @param object            ...$attributes Set of dependency injection attributes to use
      */
     public function __construct(
-        public ?string $key = null
+        public ?string $key = null,
+        public ?string $type = null,
+        public bool $nullable = false,
+        object ...$attributes,
```

Comment:
Should this be an array instead of variadic to future-proof (we someday want to add additional options to this attribute)?

Reply:
Let's make it an array, you're right about future-proofness.

Reply:
I made it `object|object[]` as I think the most common use-case will be to use one attribute here.